src/cfml/system/util/CompileDSL.cfc

```diff
@@ -0,0 +1,402 @@
+	property name='sourcePaths'				type='array';
+	property name='recursive'				type='boolean';
+	property name='createJar'				type='boolean';
+	property name='jarNameString'			type='string';
+	property name='libsDir'					type='string';
+	property name='compileOptionsString'	type='string';
+	property name='jarOptionsString'		type='string';
+    property name='javaBinFolder'           type='string';
+
+    //DI
+	property name="packageService"	inject="PackageService";
+    property name='wirebox'         inject='wirebox';
+    property name='fileSystemUtil'  inject='FileSystem';
+	property name='shell'	        inject='shell';
+    property name='job'		        inject='interactiveJob';
+	property name="tempDir" 		inject="tempDir@constants";
+
+    /*
+    have a classpath
+    have a verbose flag
+    have a encoding flag option
+    */
+
+    public function init() {
+        setSourceDirectory( 'src\main\java\' );
+        setClassOutputDirectory( 'classes\java\main\' );
+        setVerbose( false );
+        setEncode( '' );
+		setSourcePaths( [''] );
+		setRecursive( false );
+		setCreateJar( false );
+		setLibsDir( 'libs' );
+		setCompileOptionsString( '' );
+		setJarOptionsString( '' );
+		setJarNameString( 'output.jar' );
+        return this;
+    }
+
+	/**
+	 * Sets the directory to run the command in
+  	 **/
+	function projectRoot( required projectRoot ) {
+		setProjectRoot( fileSystemutil.resolvePath( projectRoot ) );
+		return this;
+	}
+
+	function fromSource( required any sourcePaths ) {
+		if( isSimpleValue( arguments.sourcePaths ) ) {
+			arguments.sourcePaths = listToArray( arguments.sourcePaths, ",", true );
+		}
+		arguments.sourcePaths = arguments.sourcePaths.map( function( s ) {
+			return fileSystemutil.resolvePath( arguments.s, getProjectRoot() );
+		} );
+		variables.sourcePaths = arguments.sourcePaths;
+		return this;
+	}
+
+    function toClasses( required classOutputDirectory ){
+        setClassOutputDirectory( fileSystemutil.resolvePath( classOutputDirectory, getProjectRoot() ) )
+        return this;
+    }
+
+    function withVerbose() {
+        setVerbose( true );
+        return this;
+    }
+
+    function withEncoding( required encodeValue ) {
+        setEncode( encodeValue );
+        return this;
+    }
+
+	function recursive() {
+		setRecursive( true );
+		return this;
+	}
+
+	function toJar( string jarName='' ) {
+		setJarNameString( jarName );
+		setCreateJar( true );
+		return this;
+	}
+
+	function libsDir( required libsDir ) {
+		setLibsDir(libsDir);
+		return this;
+	}
+
+	function compileOptions( required string options ) {
+		setCompileOptionsString( options );
+		return this;
+	}
+
+	function jarOptions( required string options) {
+		setJarOptionsString( options );
+		return this;
+	}
+
+    function run() {
+		job.start( 'Compiling' );
+
+        job.start( 'find jdk bin directory' );
+		setJavaBinFolder( findJDKBinDirectory() );
+        job.complete( getVerbose() );
```

Comment:
@BalbinoOrtus You only need to pass the verbose flag to the outermost job.  It will automatically apply to all inner jobs

Reply:
done